cudapoa/samples/sample_cudapoa.cpp

```diff
@@ -124,22 +126,127 @@ void process_batch(Batch* batch, bool msa, bool print)
     }
 }
 
-void generate_window_data(const std::string& input_file, const int number_of_windows, const int max_sequences_per_poa,
-                          std::vector<std::vector<std::string>>& windows, BatchSize& batch_size)
+size_t estimate_max_poas(const BatchSize& batch_size, const bool banded_alignment, const bool msa_flag)
 {
-    parse_window_data_file(windows, input_file, number_of_windows); // Generate windows.
-    assert(get_size(windows) > 0);
+    size_t total = 0, free = 0;
+    cudaMemGetInfo(&free, &total);
+    size_t mem_per_batch         = 0.9 * free; // Using 90% of GPU available memory for cudapoa batch.
+    const int32_t mismatch_score = -6, gap_score = -8, match_score = 8;
 
-    int32_t max_read_length = 0;
-    for (auto& window : windows)
+    int64_t sizeof_ScoreT       = 2;
+    int64_t device_size_per_poa = 0;
+
+    if (use32bitScore(batch_size, gap_score, mismatch_score, match_score))
     {
-        for (auto& seq : window)
+        sizeof_ScoreT = 4;
+        if (use32bitSize(batch_size, banded_alignment))
         {
-            max_read_length = std::max(max_read_length, get_size<int>(seq));
+            device_size_per_poa = BatchBlock<int32_t, int32_t>::compute_device_memory_per_poa(batch_size, banded_alignment, msa_flag);
         }
+        else
+        {
+            device_size_per_poa = BatchBlock<int32_t, int16_t>::compute_device_memory_per_poa(batch_size, banded_alignment, msa_flag);
```

Comment:
Since this is an implementation class, this breaks the abstraction barrier of only using public headers (i.e. those in the `include` folder)

Reply:
fixed